packages/js-moi-providers/src.ts/base-provider.ts

```diff
@@ -1242,6 +1283,31 @@ export class BaseProvider extends AbstractProvider {
         }
     }
 
+    protected processWsResult(event: ProviderEvents, result: unknown): unknown {
+        if (event === 'newPendingInteractions') {
+            if (typeof result === "string") {
+                return result.startsWith("0x") ? result : `0x${result}`;
+            }
+
+            ErrorUtils.throwError("Invalid response received", ErrorCode.SERVER_ERROR);
+        }
+
+        if (typeof event === "string" && ["newTesseracts"].includes(event)) {
+            return result;
```

Comment:
Like `event === 'newPendingInteractions'` can we add a check for  `newTesseracts`.

Reply:
Overall if possible refactor and improve the code for this function.